File: Exercise/Medium/05. Find Maximum in List/Code/Find Maximum in List.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        string line = Console.ReadLine();

        string[] tokens = line.Split(',');
        List<int> numbers = new List<int>();

        foreach (string token in tokens) {
            int num;
            if (int.TryParse(token, out num)) {
                numbers.Add(num);
            }
        }

        int max = int.MinValue;
        foreach (int num in numbers) {
            if (num > max) {
                max = num;
            }
        }

        Console.WriteLine(max);
    }
}

File: Exercise/Medium/05. Find Maximum in List/Code/Find Maximum in List.ts
function FindMax(line: string): number {
    const tokens: string[] = line.split(',');
    const numbers: number[] = tokens.map(token => Number(token.trim()));
    
    let max: number = Number.MIN_SAFE_INTEGER;
    for (let num of numbers) {
        if (num > max) {
            max = num;
        }
    }
    return max; 
}

console.log(FindMax("1, 2, 3, 4, 5")); 
